### DISCUSSION TITLE
Is it possible to crop and resize at the same time with Imagesharp.Web?

### DISCUSSION THREAD

**Discussion by xumix** [SixLabors/ImageSharp.Web#168] [Q&A] [answered]
I've tried `?width=300&rmode=crop&rxy=30,30` but looks like it does nothing except resizing.
What is the proper way? Is it possible to crop first and resize then?

**Comment by JimBobSquarePants** [comment, marked as answer]
ResizeMode.Crop crops the excess off a given set of bounds (`width` & `height`). Since you've only provided the width the image will be resized according to it's input aspect ratio which leaves nothing to crop.  The `rxy` field then denotes the center point to base any cropping on. 

Relevant API docs.
https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Processing.ResizeMode.html?q=resizemode
https://docs.sixlabors.com/api/ImageSharp.Web/SixLabors.ImageSharp.Web.Processors.ResizeWebProcessor.html#SixLabors_ImageSharp_Web_Processors_ResizeWebProcessor_Xy

If you're curious as to how we do the calculation the source code is here. 
https://github.com/SixLabors/ImageSharp/blob/af519602a56ae235ded7d76e787b88642f648de4/src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeHelper.cs#L153

P.S. In future can please use the appropriate repository (ImageSharp.Web) and topic (Q&A) for questions. I've transferred this discussion but it adds overhead that can be overwhelming when it happens often..

**Reply by xumix** [reply]
Thanks for your efforts! I've clicked the link ` Ask a Question` found here to create the question `https://github.com/SixLabors/ImageSharp.Web/issues/new/choose`
I tried setting width and height but it seem to have no effect on the image :(

**Reply by JimBobSquarePants** [reply]
>Thanks for your efforts! I've clicked the link Ask a Question found here to create the question https://github.com/SixLabors/ImageSharp.Web/issues/new/choose

Whoops! We must have a bad template then, sorry about that, I'll get it updated.

Let me have a look locally and get back to you regarding the resize/crop.

**Reply by JimBobSquarePants** [reply]
`rmode=crop` is the default mode. I just used the sample site to check everything works and the result is as expected

![image](https://user-images.githubusercontent.com/385879/122632118-7d59b600-d0c8-11eb-992e-d2c3d4e40d33.png)

**Reply by xumix** [reply]
> 
> 
> `rmode=crop` is the default mode. I just used the sample site to check everything works and the result is as expected
> 
> ![image](https://user-images.githubusercontent.com/385879/122632118-7d59b600-d0c8-11eb-992e-d2c3d4e40d33.png)

Thanks for your efforts! Looks lie you've missed the crucial part with `rxy=30,30`
The desired behavior is to crop first with an anchor and then resize the image

**Reply by JimBobSquarePants** [reply]
That’s not how it works. `rxy` is a centre point for any cropping as a result of cropping excess from a given aspect ratio.

If you want separate cropping you have to implement that processor yourself.

**Comment by ronaldbarendse** [comment]
@xumix Using `?width=300&rmode=crop&rxy=30,30` only resizes, because of two things:
1. Only specifying a single dimension will use the aspect ratio of the original image, so there's nothing to crop;
2. The `rxy` command expects a percentage based coordinate (values between 0-1, like `rxy=0.5,0.5`).

If you want to crop a specific rectangle from an image and resize that, you'll indeed need to implement that processor yourself and make sure that's added before the default `ResizeWebProcessor` (see https://github.com/SixLabors/ImageSharp.Web/issues/182). We needed to do the same in Umbraco, which uses a percentage based crop rectangle (see https://github.com/umbraco/Umbraco-CMS/pull/10623 and the [`CropWebProcessor`](https://github.com/umbraco/Umbraco-CMS/blob/5bfab13dc5a268714aad2426a2b68ab5561a6407/src/Umbraco.Web.Common/ImageProcessors/CropWebProcessor.cs)).
